setup/_pattoo/systemd.py

```diff
@@ -275,25 +299,27 @@ def run_systemd():
         None
 
     """
-    if getpass.getuser() != 'travis':
-        print('Loading system daemon configurations')
-        shared.run_script('sudo systemctl daemon-reload')
-        print('Enabling system daemons')
-        shared.run_script('sudo systemctl enable pattoo_apid')
-        shared.run_script('sudo systemctl enable pattoo_api_agentd')
-        shared.run_script('sudo systemctl enable pattoo_ingesterd')
-        print('Starting system daemons')
-        shared.run_script('sudo systemctl start pattoo_apid')
-        shared.run_script('sudo systemctl start pattoo_api_agentd')
-        shared.run_script('sudo systemctl start pattoo_ingesterd')
-    else:
+    # Say what we are doing
+    print('Checking if daemons are already running')
+
+    # Loop through daemon list
+    for daemon in daemons:
+        command = '''
+systemctl is-active {} daemon --quiet service-name'''.format(daemon)
+
+        # Check status code of daemon
+        status = os.system(command)
```

Comment:
Avoid using os.system as a rule, modify runscript to figure out the exit code. The advantage is that runscript will also give you access to the error messages that os.system won't. Make runscript return an exit code for example, if it doesn't already. Consider a failsafe parameter for runscript so that it doesn't die under certain conditions, if it doesn't already.